### ISSUE TITLE
Document Crosby's branch cleanup report behavior

### ISSUE THREAD

**Issue by alintawalshy66** [alintawalshy66/resources#54]
## Scope

Update Crosby documentation to describe the blocking branch cleanup report.

## Blocked By

- #53

## User Stories Covered

- As a Crosby user, I want branch preflight failures to explain what branches exist and what I should do with them, so I can unblock myself quickly.
- As a Crosby user, I want the preflight to remain blocking, so Crosby does not start from an unsafe or unclear branch state.
- As a Crosby user, I want the report to avoid destructive commands, so cleanup remains an explicit manual decision.

## Expected Files

- `shared-workflows/pi/extensions/crosby/README.md`

## Do Not Touch

- `shared-workflows/pi/extensions/crosby/index.ts`
- `shared-workflows/pi/extensions/crosby/lib-v2.mjs`
- `shared-workflows/pi/extensions/crosby/*.test.mjs`
- `shared-workflows/portable/skills/`

## Crosby Locks

- `shared-workflows/pi/extensions/crosby/README.md`

## Test Command

```bash
rg -n "branch cleanup|Base branch: main|No branches were changed|local branches" shared-workflows/pi/extensions/crosby/README.md
```

## Acceptance Criteria

- [x] README describes the branch cleanup report.
- [x] README states that `main` is the base branch.
- [x] README states that only local branches are reported.
- [x] README states that Crosby does not modify branches.
- [x] README avoids deletion command examples.

## Notes

Parent: #50
